x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequestSenderTests.java

```diff
@@ -179,10 +182,11 @@ public void testRetryThenSuccess() {
             throw new AssertionError(e);
         }
         assertThat(sent, hasSize(5));
-        var firstRound = groupRequests(sent, 3);
-        assertThat(firstRound, equalTo(Map.of(node1, List.of(shard1, shard5), node4, List.of(shard2), node2, List.of(shard3, shard4))));
-        var secondRound = groupRequests(sent, 2);
-        assertThat(secondRound, equalTo(Map.of(node2, List.of(shard2), node3, List.of(shard5))));
+        assertThat(
+            take(sent, 3),
+            containsInAnyOrder(nodeRequest(node1, shard1, shard5), nodeRequest(node4, shard2), nodeRequest(node2, shard3, shard4))
+        );
+        assertThat(take(sent, 2), containsInAnyOrder(nodeRequest(node2, shard2), nodeRequest(node3, shard5)));
```

Comment:
Lets compare domain objects rather than maps of lists

Reply:
FWIW you can do `matchesMap()` and get lovely error messaging. If you want. But I have no objections to comparing any way you like.